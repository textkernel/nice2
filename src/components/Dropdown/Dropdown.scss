.Dropdown {
    display: inline-block;
    position: relative;
    font: {
        family: var(--font-family-primary);
        size: var(--font-size-normal);
    }
}
