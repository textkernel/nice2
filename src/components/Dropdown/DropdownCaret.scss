.DropdownCaret {
    transition: fill var(--transition-duration);
}
