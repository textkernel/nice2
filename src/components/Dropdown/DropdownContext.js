import React from 'react';

const DropdownContext = React.createContext({});

export const DropdownProvider = DropdownContext.Provider;
export const DropdownConsumer = DropdownContext.Consumer;
