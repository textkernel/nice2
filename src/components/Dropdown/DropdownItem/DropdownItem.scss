.DropdownItem {
    color: var(--color-foreground);
    cursor: pointer;
    display: block;
    line-height: var(--line-height-normal);
    padding: var(--spacing-2x) var(--spacing-3x);
    user-select: none;
    white-space: nowrap;

    &:hover {
        background-color: var(--color-neutral-25);
    }

    &--checkbox {
        padding: 0;
    }

    &--disabled {
        cursor: not-allowed;
        color: var(--color-neutral-75);

        &:active {
            background-color: var(--color-neutral-25);
        }
    }
    &--isStatic {
        background-color: var(--color-background);
        color: var(--color-neutral-75);
        cursor: default;

        &:hover,
        &:active {
            background-color: var(--color-background);
        }
    }

    &__checkbox {
        .Checkbox__label {
            padding: var(--spacing-2x) var(--spacing-3x);
            line-height: var(--line-height-normal);
        }
    }
}
