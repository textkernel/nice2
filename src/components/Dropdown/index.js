export { default as Dropdown } from './Dropdown';
export { default as DropdownFilter } from './DropdownFilter';
export { default as DropdownGroup } from './DropdownGroup';
export { default as DropdownItem } from './DropdownItem';
export { DropdownConsumer } from './DropdownContext';
