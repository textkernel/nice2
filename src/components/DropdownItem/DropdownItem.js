import React from 'react';
import PropTypes from 'prop-types';
import bem from '../../packages/bem';
import styles from './DropdownItem.scss';

const { block } = bem({
    name: 'DropdownItem',
    classnames: styles,
    propsToMods: ['disabled', 'isStatic']
});

const DropdownItem = props => {
    const { checkbox, children, disabled, isStatic, ...rest } = props;

    if (checkbox) {
        return (
            <div {...rest} {...block(props)}>
                <input type="checkbox" />
                <label>
                    {children}
                </label>
            </div>
        );
    }

    return (
        <div {...rest} {...block(props)}>
            {children}
        </div>
    );
};

DropdownItem.propTypes = {
    checkbox: PropTypes.bool,
    /** Item label */
    children: PropTypes.node.isRequired,
    /** Whether the item is disabled */
    disabled: PropTypes.bool,
    /** Callback function, fired when item is clicked */
    onClick: PropTypes.func,
    /** If true, this item does not allow user interaction */
    isStatic: PropTypes.bool,
    /** Selected value of this item */
    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
};

DropdownItem.defaultProps = {
    checkbox: false,
    disabled: false,
    onClick: null,
    isStatic: false,
    value: null
};

export default DropdownItem;
