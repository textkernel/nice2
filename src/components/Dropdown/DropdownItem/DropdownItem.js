import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import Checkbox from '../../Checkbox';
import Text from '../../Text';
import { DropdownConsumer } from '../DropdownContext';
import styles from './DropdownItem.scss';

const { block, elem } = bem({
    name: 'DropdownItem',
    classnames: styles,
    propsToMods: []
});

const DropdownItem = props => {
    const { children } = props;

    return (
        <DropdownConsumer>
            {({ multiselect }) => (
                <div {...block(props)}>
                    {multiselect ? (
                        <div {...elem('multi', props)}>
                            <Checkbox>{children}</Checkbox>
                        </div>
                    ) : (
                        <div {...elem('single', props)}>
                            <Text inline>{children}</Text>
                        </div>
                    )}
                </div>
            )}
        </DropdownConsumer>
    );
};

DropdownItem.displayName = 'DropdownItem';

DropdownItem.propTypes = {
    disabled: PropTypes.bool
};

DropdownItem.defaultProps = {
    disabled: false
};

export default DropdownItem;
