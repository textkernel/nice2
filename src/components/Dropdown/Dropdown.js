import React, { cloneElement, createRef, Children, Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import diacritics from 'diacritics';
import bem from 'bem';
import { escapeRegExp } from '../../utils';
import Button from '../Button';
import Input from '../Input';
import DropdownCaret from './DropdownCaret';
import DropdownItem from './DropdownItem';
import styles from './Dropdown.scss';
import { CONTEXTS, SIZES } from '../../constants';

class Dropdown extends PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            expanded: props.initiallyOpened,
            filterValue: null,
            fromRight: false,
            fromRightOffset: 0,
            selection: props.value
        };

        this.dropdown = createRef();
        this.filter = createRef();
        this.list = createRef();
        this.handleChange = this.handleChange.bind(this);
        this.handleClickOutside = this.handleClickOutside.bind(this);
        this.handleEscPress = this.handleEscPress.bind(this);
        this.toggleDropdown = this.toggleDropdown.bind(this);
        window.addEventListener('click', this.handleClickOutside, false);
        window.addEventListener('keyup', this.handleEscPress, true);
        window.addEventListener('resize', this.determineListOrientation.bind(this), false);
    }

    componentDidMount() {
        this.determineListOrientation();
    }

    componentWillUnmount() {
        window.removeEventListener('click', this.handleClickOutside, false);
        window.removeEventListener('keyup', this.handleEscPress, true);
        window.removeEventListener('resize', this.determineListOrientation.bind(this), false);
    }

    determineListOrientation() {
        if (!this.list || !this.list.current) {
            return true;
        }

        const windowWidth = Math.max(document.documentElement.clientWidth, window.innerWidth || 0);

        const { left, right, width } = this.list.current.getBoundingClientRect();
        const { fromRight, fromRightOffset } = this.state;

        if (left - width > 0 && right >= windowWidth && !fromRight) {
            this.setState({
                fromRight: true,
                fromRightOffset: right
            });
        } else if (fromRight && fromRightOffset < windowWidth) {
            this.setState({
                fromRight: false,
                fromRightOffset: 0
            });
        }

        return true;
    }

    focusFilter() {
        // Focus filter input if enabled and dropdown is a multiselect
        const { filter, multiple } = this.props;
        if (!filter || !this.filter || !this.filter.current || !multiple) {
            return false;
        }
        this.filter.current.focus();
        return true;
    }

    handleChange(value) {
        const { multiple } = this.props;
        let newSelection;

        if (multiple) {
            const { selection } = this.state;
            // Clone existing selection or start with empty one
            newSelection = selection ? [...selection] : [];

            const index = newSelection.indexOf(value);
            if (index > -1) {
                // Value already selected: remove it
                newSelection.splice(index, 1);
            } else {
                // Add value to selection
                newSelection = [...newSelection, value];
            }
        } else {
            // Single-select dropdown: replace value
            newSelection = value;
        }

        this.setState(
            {
                selection: newSelection
            },
            () => {
                const { onChange } = this.props;
                this.focusFilter();
                if (!onChange) {
                    return true;
                }
                return onChange(newSelection);
            }
        );
    }

    handleClickOutside(e) {
        // Collapse dropdown on click outside
        if (!this.dropdown || !this.dropdown.current) {
            return false;
        }
        if (this.dropdown.current.contains(e.target)) {
            return false;
        }
        const { expanded } = this.state;
        if (!expanded) {
            return false;
        }
        this.toggleDropdown(null, true, true);
        return true;
    }

    handleEscPress(e) {
        // Collapse dropdown on esc press
        e.stopPropagation();
        const key = e.keyCode || e.which;

        if (key !== 27) {
            return false;
        }

        this.toggleDropdown(null, true, true);
        return false;
    }

    highlightLabel(label) {
        // Highlight parts of a label that match with current filter value
        const { filterValue } = this.state;

        if (!filterValue) {
            return label;
        }

        const parts = diacritics
            .remove(label)
            .split(new RegExp(`(${escapeRegExp(filterValue)})`, 'gi'));
        let pos = 0;

        return (
            <Fragment>
                {parts.map((part, i) => {
                    const match = part.toLowerCase() === filterValue.toLowerCase();
                    const originalPart = label.substring(pos, pos + part.length);
                    pos += part.length;

                    if (match) {
                        return (
                            <strong key={i.toString()} {...this.elem('highlight')}>
                                {originalPart}
                            </strong>
                        );
                    }
                    return <span key={i.toString()}>{originalPart}</span>;
                })}
            </Fragment>
        );
    }

    toggleDropdown(e, collapse = false) {
        const { expanded } = this.state;

        if (e && e.stopPropagation) {
            e.stopPropagation();
        }

        const newState = {
            expanded: collapse ? false : !expanded
        };

        if (!newState.expanded) {
            newState.filterValue = null;
        }

        this.setState(newState, () => {
            const { onClose } = this.props;
            const { selection } = this.state;
            if (!newState.expanded && onClose) {
                onClose(selection);
            }
        });

        return true;
    }

    extendItemProps(children) {
        // Clone dropdown item and extend its props with those
        // needed for a multiselect dropdown item
        const { multiple } = this.props;
        const { selection } = this.state;

        return Children.map(children, (item, index) =>
            cloneElement(item, {
                checkbox: multiple,
                checked: multiple && (selection || []).indexOf(item.props.value) > -1,
                children: this.highlightLabel(item.props.children),
                index,
                onSelect: this.handleChange
            })
        );
    }

    filteredChildren() {
        const { children, filter } = this.props;
        const { filterValue } = this.state;

        if (!filter || !filterValue) {
            return this.extendItemProps(children);
        }

        const re = new RegExp(
            `${filter.matchPosition === 'start' ? '^' : ''}(${escapeRegExp(filterValue)})`,
            `g${!filter.matchCase ? 'i' : ''}`
        );

        const filteredChildren = Children.toArray(children).filter(item => {
            const { children: label } = item.props;
            return diacritics.remove(label).match(re);
        });

        if (!filteredChildren.length) {
            return (
                <DropdownItem isStatic>{filter.noMatchLabel || 'No matches found'}</DropdownItem>
            );
        }

        return this.extendItemProps(filteredChildren);
    }

    render() {
        const {
            children,
            context,
            heading,
            disabled,
            filter,
            initiallyOpened,
            isBlock,
            label,
            maxHeight,
            minWidth,
            size,
            ...rest
        } = this.props;
        const { expanded, filterValue } = this.state;

        return (
            <div {...rest} {...this.block()} ref={this.dropdown}>
                <Button
                    context={context}
                    disabled={disabled}
                    isBlock={isBlock}
                    onClick={this.toggleDropdown}
                    size={size}
                    {...this.elem('button')}
                >
                    {label}
                    <DropdownCaret {...this.elem('caret')} />
                </Button>
                {!!expanded &&
                    !disabled && (
                        <div {...this.elem('list')} ref={this.list}>
                            {!!heading && <div {...this.elem('heading')}>{heading}</div>}
                            {!!filter && (
                                <div {...this.elem('filter')}>
                                    <Input
                                        autoFocus
                                        isBlock
                                        onChange={e => {
                                            const { value } = e.target;

                                            this.setState({
                                                filterValue: diacritics.remove(value)
                                            });
                                        }}
                                        placeholder={filter.placeholder || null}
                                        ref={this.filter}
                                        value={filterValue || ''}
                                    />
                                </div>
                            )}
                            <div
                                style={{
                                    maxHeight,
                                    minWidth
                                }}
                                {...this.elem('list-items')}
                                role="menu"
                            >
                                {this.filteredChildren()}
                            </div>
                        </div>
                    )}
            </div>
        );
    }
}

Dropdown.propTypes = {
    /** The dropdown items */
    children: PropTypes.node,
    /** Dropdown context */
    context: PropTypes.oneOf(['link', ...CONTEXTS]),
    /** heading to be rendered at the top of dropdown list */
    heading: PropTypes.node,
    /** Whether the dropdown is disabled */
    disabled: PropTypes.bool,
    /** Filter dropdown options and highlight matches */
    filter: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.shape({
            /** If the filter is case sensitive */
            matchCase: PropTypes.bool,
            /** Whether to match at any position in the string or from the start */
            matchPosition: PropTypes.oneOf(['any', 'start']),
            /** Text to show when filter produced no matches */
            noMatchLabel: PropTypes.node,
            /** Input placeholder when filter is empty */
            placeholder: PropTypes.string
        })
    ]),
    /** Open the dropdown after first render */
    initiallyOpened: PropTypes.bool,
    /** Render dropdown as block-level element */
    isBlock: PropTypes.bool,
    /** Label for the dropdown trigger */
    label: PropTypes.string.isRequired,
    /** Max. height of dropdown list (will scroll if exceeded) */
    maxHeight: PropTypes.number,
    /** Min. width of dropdown list */
    minWidth: PropTypes.number,
    /** Whether to render a multiselect dropdown (value should be an array) */
    multiple: PropTypes.bool,
    /** Callback function that is fired upon item selection
        The current selection is passed as second argument
        In case of a multiselect, the new value is added if not yet present
        or removed if already present */
    onChange: PropTypes.func,
    /** Callback function that is fired after dropdown collapses
     Current selection is passed as first argument */
    onClose: PropTypes.func,
    /** Size of the dropdown trigger */
    size: PropTypes.oneOf(SIZES),
    /** Current dropdown value. In case of a multiselect this is expected
        to be an array of values. Otherwise it holds the value of a selected
        dropdown item. If provided, the dropdown is a controlled component */
    value: PropTypes.oneOfType([
        PropTypes.arrayOf([PropTypes.string, PropTypes.number]),
        PropTypes.string,
        PropTypes.number
    ])
};

Dropdown.defaultProps = {
    children: null,
    context: 'neutral',
    heading: null,
    disabled: false,
    filter: false,
    initiallyOpened: false,
    isBlock: false,
    maxHeight: null,
    minWidth: null,
    multiple: false,
    onChange: null,
    onClose: null,
    size: 'normal',
    value: null
};

Dropdown.propsToMods = ['context', 'disabled', 'isBlock', 'size'];
Dropdown.stateToMods = ['expanded', 'fromRight'];

export default bem(styles)(Dropdown);
