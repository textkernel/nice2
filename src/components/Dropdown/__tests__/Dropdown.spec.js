import React from 'react';
import toJson from 'enzyme-to-json';
import Dropdown from '../Dropdown';
import DropdownItem from '../DropdownItem';

describe('<Dropdown> that renders a dropdown element', () => {
    it('should render a basic dropdown correctly', () => {
        const onChange = jest.fn();
        const onClose = jest.fn();
        const wrapper = mount(
            <Dropdown label="My dropdown" onChange={onChange} onClose={onClose}>
                <DropdownItem value="1">Some item</DropdownItem>
                <DropdownItem value="2">Another item</DropdownItem>
                <DropdownItem value="3">More items</DropdownItem>
            </Dropdown>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.state().selection).toBeNull();

        // Expand the dropdown
        wrapper.find('Button').simulate('click');
        expect(wrapper.state().expanded).toBe(true);

        // Select an item and make sure state is updated and callback fired
        wrapper
            .find(DropdownItem)
            .first()
            .simulate('click');
        expect(onChange).toHaveBeenCalledWith('1');
        expect(wrapper.state().selection).toBe('1');

        // Verify that callback is fired on dropdown collapse
        wrapper.find('Button').simulate('click');
        expect(onClose).toHaveBeenCalledWith('1');
        expect(wrapper.state().expanded).toBe(false);
    });

    it('should add classes when props are changed', () => {
        const wrapper = shallow(
            <Dropdown
                label="My dropdown"
                context="primary"
                size="large"
                disabled
                isBlock
                initiallyOpened
                maxHeight={100}
                minWidth={300}
            >
                <DropdownItem value="1" disabled>
                    Some item
                </DropdownItem>
                <DropdownItem value="2">Another item</DropdownItem>
                <DropdownItem value="3">More items</DropdownItem>
            </Dropdown>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should filter items correctly', () => {
        const wrapper = mount(
            <Dropdown
                label="My dropdown"
                filter={{
                    placeholder: 'Test filtering options'
                }}
            >
                <DropdownItem value="1">test capitalization</DropdownItem>
                <DropdownItem value="2">Téšṫ diacritics</DropdownItem>
                <DropdownItem value="3">Match positioning test</DropdownItem>
            </Dropdown>
        );
        wrapper.setState({
            expanded: true
        });
        expect(toJson(wrapper)).toMatchSnapshot();

        const filterInput = wrapper.find('input');
        filterInput.simulate('change', { target: { value: 'Test' } });
        expect(wrapper.find(DropdownItem).length).toBe(3);

        wrapper.setProps({
            filter: {
                matchCase: true,
                matchPosition: 'start'
            }
        });
        filterInput.simulate('change', { target: { value: 'test' } });
        expect(wrapper.find(DropdownItem).length).toBe(1);
    });
});
