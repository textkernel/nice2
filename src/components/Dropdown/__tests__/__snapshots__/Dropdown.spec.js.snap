// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown> that renders a dropdown element should add classes when props are changed 1`] = `
<div
  className="Dropdown Dropdown--context_primary Dropdown--disabled Dropdown--isBlock Dropdown--size_large Dropdown--expanded"
>
  <Button
    className="Dropdown__button Dropdown__button--context_primary Dropdown__button--disabled Dropdown__button--isBlock Dropdown__button--size_large Dropdown__button--expanded"
    context="primary"
    disabled={true}
    isBlock={true}
    onClick={[Function]}
    size="large"
    type="button"
  >
    My dropdown
    <DropdownCaret
      className="Dropdown__caret Dropdown__caret--context_primary Dropdown__caret--disabled Dropdown__caret--isBlock Dropdown__caret--size_large Dropdown__caret--expanded"
    />
  </Button>
</div>
`;

exports[`<Dropdown> that renders a dropdown element should filter items correctly 1`] = `
<Dropdown
  context="neutral"
  disabled={false}
  filter={
    Object {
      "placeholder": "Test filtering options",
    }
  }
  heading={null}
  initiallyOpened={false}
  isBlock={false}
  label="My dropdown"
  maxHeight={null}
  minWidth={null}
  multiple={false}
  onChange={null}
  onClose={null}
  size="normal"
  value={null}
>
  <div
    className="Dropdown Dropdown--context_neutral Dropdown--size_normal Dropdown--expanded"
  >
    <Button
      className="Dropdown__button Dropdown__button--context_neutral Dropdown__button--size_normal Dropdown__button--expanded"
      context="neutral"
      disabled={false}
      isBlock={false}
      onClick={[Function]}
      size="normal"
      type="button"
    >
      <button
        className="Button Button--context_neutral Button--size_normal Dropdown__button Dropdown__button--context_neutral Dropdown__button--size_normal Dropdown__button--expanded"
        context="neutral"
        disabled={false}
        onClick={[Function]}
        size="normal"
        type="button"
      >
        My dropdown
        <DropdownCaret
          className="Dropdown__caret Dropdown__caret--context_neutral Dropdown__caret--size_normal Dropdown__caret--expanded"
        >
          <svg
            className="DropdownCaret Dropdown__caret Dropdown__caret--context_neutral Dropdown__caret--size_normal Dropdown__caret--expanded"
            viewBox="0 0 292.362 292.362"
          >
            <g>
              <path
                d="M286.935,69.377c-3.614-3.617-7.898-5.424-12.848-5.424H18.274c-4.952,0-9.233,1.807-12.85,5.424   C1.807,72.998,0,77.279,0,82.228c0,4.948,1.807,9.229,5.424,12.847l127.907,127.907c3.621,3.617,7.902,5.428,12.85,5.428   s9.233-1.811,12.847-5.428L286.935,95.074c3.613-3.617,5.427-7.898,5.427-12.847C292.362,77.279,290.548,72.998,286.935,69.377z"
              />
            </g>
          </svg>
        </DropdownCaret>
      </button>
    </Button>
    <div
      className="Dropdown__list Dropdown__list--context_neutral Dropdown__list--size_normal Dropdown__list--expanded"
    >
      <div
        className="Dropdown__filter Dropdown__filter--context_neutral Dropdown__filter--size_normal Dropdown__filter--expanded"
      >
        <ForwardRef
          autoFocus={true}
          context="brand"
          disabled={false}
          isBlock={true}
          onChange={[Function]}
          placeholder="Test filtering options"
          size="normal"
          type="text"
          value=""
        >
          <input
            autoFocus={true}
            className="Input Input--context_brand Input--isBlock Input--size_normal"
            disabled={false}
            onChange={[Function]}
            placeholder="Test filtering options"
            type="text"
            value=""
          />
        </ForwardRef>
      </div>
      <div
        className="Dropdown__list-items Dropdown__list-items--context_neutral Dropdown__list-items--size_normal Dropdown__list-items--expanded"
        role="menu"
        style={
          Object {
            "maxHeight": null,
            "minWidth": null,
          }
        }
      >
        <DropdownItem
          checkbox={false}
          checked={false}
          disabled={false}
          index={0}
          isStatic={false}
          key=".0"
          onClick={null}
          onSelect={[Function]}
          value="1"
        >
          <div
            className="DropdownItem"
            onClick={[Function]}
            role="menuitem"
          >
            test capitalization
          </div>
        </DropdownItem>
        <DropdownItem
          checkbox={false}
          checked={false}
          disabled={false}
          index={1}
          isStatic={false}
          key=".1"
          onClick={null}
          onSelect={[Function]}
          value="2"
        >
          <div
            className="DropdownItem"
            onClick={[Function]}
            role="menuitem"
          >
            Téšṫ diacritics
          </div>
        </DropdownItem>
        <DropdownItem
          checkbox={false}
          checked={false}
          disabled={false}
          index={2}
          isStatic={false}
          key=".2"
          onClick={null}
          onSelect={[Function]}
          value="3"
        >
          <div
            className="DropdownItem"
            onClick={[Function]}
            role="menuitem"
          >
            Match positioning test
          </div>
        </DropdownItem>
      </div>
    </div>
  </div>
</Dropdown>
`;

exports[`<Dropdown> that renders a dropdown element should render a basic dropdown correctly 1`] = `
<Dropdown
  context="neutral"
  disabled={false}
  filter={false}
  heading={null}
  initiallyOpened={false}
  isBlock={false}
  label="My dropdown"
  maxHeight={null}
  minWidth={null}
  multiple={false}
  onChange={[MockFunction]}
  onClose={[MockFunction]}
  size="normal"
  value={null}
>
  <div
    className="Dropdown Dropdown--context_neutral Dropdown--size_normal"
  >
    <Button
      className="Dropdown__button Dropdown__button--context_neutral Dropdown__button--size_normal"
      context="neutral"
      disabled={false}
      isBlock={false}
      onClick={[Function]}
      size="normal"
      type="button"
    >
      <button
        className="Button Button--context_neutral Button--size_normal Dropdown__button Dropdown__button--context_neutral Dropdown__button--size_normal"
        context="neutral"
        disabled={false}
        onClick={[Function]}
        size="normal"
        type="button"
      >
        My dropdown
        <DropdownCaret
          className="Dropdown__caret Dropdown__caret--context_neutral Dropdown__caret--size_normal"
        >
          <svg
            className="DropdownCaret Dropdown__caret Dropdown__caret--context_neutral Dropdown__caret--size_normal"
            viewBox="0 0 292.362 292.362"
          >
            <g>
              <path
                d="M286.935,69.377c-3.614-3.617-7.898-5.424-12.848-5.424H18.274c-4.952,0-9.233,1.807-12.85,5.424   C1.807,72.998,0,77.279,0,82.228c0,4.948,1.807,9.229,5.424,12.847l127.907,127.907c3.621,3.617,7.902,5.428,12.85,5.428   s9.233-1.811,12.847-5.428L286.935,95.074c3.613-3.617,5.427-7.898,5.427-12.847C292.362,77.279,290.548,72.998,286.935,69.377z"
              />
            </g>
          </svg>
        </DropdownCaret>
      </button>
    </Button>
  </div>
</Dropdown>
`;
